src/app/dataset-view/additional-components/metadata-component/components/add-push-source/add-push-source.component.ts

```diff
@@ -76,11 +71,62 @@ export class AddPushSourceComponent extends BaseMainEventComponent {
         }),
     });
 
-    public get readForm(): FormGroup {
+    public initEditForm(): void {
+        this.trackSubscription(
+            this.editService
+                .getEventAsYaml(this.getDatasetInfoFromUrl(), this.queryParamName ?? "")
+                .subscribe((result) => {
+                    this.history = this.editService.history;
+                    if (this.unsupportedSourceName()) {
+                        this.zone.run(() => {
+                            this.navigationServices.navigateToPageNotFound();
+                        });
```

Comment:
Can you explain why this requires a different execution zone?

Reply:
Still deleted it. The test gave a warning, I changed the test and the warning disappeared